kythe/cxx/extractor/bazel_artifact.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2020 The Kythe Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef KYTHE_CXX_EXTRACTOR_BAZEL_ARTIFACT_H_
+#define KYTHE_CXX_EXTRACTOR_BAZEL_ARTIFACT_H_
+
+#include <string>
+#include <tuple>
+#include <vector>
+
+namespace kythe {
+
+/// \brief A pair of local path and canonical URI for a given Bazel output file.
+/// Bazel typically provides both a workspace-local path to a file as well as a
+/// canonical URI and both can be useful to clients.
+struct BazelArtifactFile {
```

Comment:
Just curious, this sort of smells like a map. Should it be?

Reply:
I don't see how using a map would be beneficial.  While I wouldn't expect duplicate paths in a list of artifacts:

1) The URI isn't unique (`data:` or another content-addressable scheme may be used)
2) The unique-ness of the file paths isn't really a core part of the semantics
3) The resulting API is less readable (`std::pair` is *terrible*), so without a compelling case for #2, the decreased readability of the resulting API obtains (IMO).